src/helpers/GoalController.ts

```diff
@@ -72,6 +93,7 @@ export const modifyGoal = async (
     goalColor,
   });
   sendUpdatedGoal(goalId, ancestors);
+  await updateHint(goalTags.id, goalHint);
```

Comment:
can we make it async ?

Reply:
@Tushar-4781 Which function are you referring to: "sendUpdatedGoal" or "updateHint"?

Reply:
Both

Reply:
@vinaybadgujar102 please make them async

Reply:
@vinaybadgujar102 this conversation is not resolved please make them async

Reply:
@Tushar-4781 From my understanding, the code is already asynchronous using 'await'. Can you provide more details on the suggested changes?

Reply:
Utilizing await will pause subsequent execution, consequently impeding the progress of goal updates. Employing then and catch for each function allows them to operate asynchronously in the background, as there is no reliance on their outcomes, enabling independent execution in the background

Reply:
```
const sendUpdatedGoalPromise = sendUpdatedGoal(goalId, ancestors);
const updateHintPromise = updateHint(goalTags.id, goalHint);
await Promise.all([sendUpdatedGoalPromise, updateHintPromise]);
```
@Tushar-4781 will this work?

Reply:
@vinaybadgujar102 
1. use allSettled
2. remove the await and use catch for handling the error.

Reply:
@Tushar-4781 I have updated the code, please review